### DISCUSSION TITLE
How to use atoti on Google Colab?

### DISCUSSION THREAD

**Discussion by bloussou** [atoti/atoti#473] [Q&A] [answered]
Hello,
I want to do use atoti on google colab, how can I proceed?
Thanks

**Comment by bloussou** [comment, marked as answer]
It is not possible to use atoti-jupyterlab in colab, but atoti can work without its jupyterlab extension as follow:

```
!pip install atoti
!pip install pyngrok
```

```python
import atoti as tt
from pyngrok import ngrok
import pandas as pd

port = 9090
ngrok_connection = ngrok.connect(port)

session = tt.create_session(config={"port": port})

table = session.read_pandas(
    pd.DataFrame({"ID": [1,2,3,4], "Product": ["a", "b","c","d"], "Amount": [10.0, 20.0, 30.0, 40.0]}),
    table_name = "sales"
)
cube = session.create_cube(table)

print(ngrok_connection.public_url)
```

`pyngrok` makes possible the access to the atoti app to make interactive dashboards.

**Reply by mkj256-mks** [reply]
Hello, i tried the code above in a Microsoft Fabric notebook, where it is also impossible to run atoti-jupyterlab, but i am getting the error 4018.

i am not familar with pyngrok, but would the port = 9090 something that i should use by default, or not that is something you got from somewhere, and it was only assigned for your case? Thanks

**Comment by bloussou** [comment]
After more work on this :
- `pyngrok` free usage limitations (see [pricing](https://ngrok.com/pricing)) is not enough to have a good experience with the atoti app.
- We tried to use [colab advanced outputs](https://colab.research.google.com/notebooks/snippets/advanced_outputs.ipynb) but colab does not support websockets (see this [issue](https://discourse.bokeh.org/t/colabs-bokeh-server/5286/2)) therefore it is not possible to use the atoti app with this technique.
- Third parties cookies will have to be authorized for atoti to work properly on Colab with advanced outputs.

Other users, please comment here if you use Colab.

**Reply by rcolle** [reply]
> pyngrok free usage limitations (see [pricing](https://ngrok.com/pricing)) is not enough to have a good experience with the atoti app.

Using the latest atoti version with the revamped UI, the experience is actually quite good.
It's likely because it's now loading plugins and charts on demand, not overloading ngrok at startup with too many requets.

**Comment by sumitrathi95** [comment]
Hello Team,

When I try to use cube.schema command it is showing below error:

`Missing Graphviz library which is required to display the graph. It can be installed with Conda: conda install graphviz or by following the download instructions at https://www.graphviz.org/download/.`

![image](https://user-images.githubusercontent.com/58877289/120597403-9d4f6f80-c462-11eb-9f82-2b84db5f8eb8.png)
I have installed graphviz. pydot, pygraphviz in colab still its failing.

**Comment by tibdex** [comment]
Hi @sumitrathi95, how did you install `graphviz` in your environment? Did you try restarting the kernel and the atoti session after installing it? Maybe this would help: https://jakevdp.github.io/blog/2017/12/05/installing-python-packages-from-jupyter/.

**Comment by Lexora** [comment]
What about using https://github.com/antoniomika/sish instead of pyngrok @bloussou ?
